packages/realm-server/handlers/handle-fetch-user.ts

```diff
@@ -0,0 +1,177 @@
+import { SupportedMimeType } from '@cardstack/runtime-common';
+import Koa from 'koa';
+import { sendResponseForSystemError, setContextResponse } from '../middleware';
+import { RealmServerTokenClaim } from '../utils/jwt';
+import {
+  getMostRecentSubscription,
+  getMostRecentSubscriptionCycle,
+  getPlan,
+  getUserByMatrixUserId,
+  Plan,
+  Subscription,
+  SubscriptionCycle,
+  sumUpCreditsLedger,
+} from '../billing/billing-queries';
+import { CreateRoutesArgs } from '../routes';
+
+type FetchUserResponse = {
+  data: {
+    type: 'user';
+    id: string;
+    attributes: {
+      matrixUserId: string;
+      stripeCustomerId: string;
+      creditsAvailableInPlanAllowance: number;
+      extraCreditsAvailableInBalance: number;
+    };
+    relationships: {
+      subscription: {
+        data: {
+          type: 'subscription';
+          id: string;
+        };
+      } | null;
+    };
+  } | null;
+  included:
+    | [
+        {
+          type: 'subscription';
+          id: string;
+          attributes: {
+            startedAt: number;
+            endedAt: number | null;
+            status: string;
+          };
+          relationships: {
+            plan: {
+              data: {
+                type: 'plan';
+                id: string;
+              };
+            };
+          };
+        },
+        {
+          type: 'plan';
+          id: string;
+          attributes: {
+            name: string;
+            monthlyPrice: number;
+            creditsIncluded: number;
+          };
+        },
+      ]
+    | null;
+};
+
+export default function handleFetchUserRequest({
+  dbAdapter,
+}: CreateRoutesArgs): (ctxt: Koa.Context, next: Koa.Next) => Promise<void> {
+  return async function (ctxt: Koa.Context, _next: Koa.Next) {
+    let token = ctxt.state.token as RealmServerTokenClaim;
+    if (!token) {
+      await sendResponseForSystemError(
+        ctxt,
+        'token is required to create realm',
+      );
+      return;
+    }
+
+    let { user: matrixUserId } = token;
+    let user = await getUserByMatrixUserId(dbAdapter, matrixUserId);
+    let mostRecentSubscription: Subscription | null = null;
+    if (user) {
```

Comment:
Maybe return a 404 here? This way you won't have to have user conditions in your responses